### DISCUSSION TITLE
Hardware help - totally confused!

### DISCUSSION THREAD

**Discussion by metaljay** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#621] [General]
This truly has me stumped...

I have made about 20x of these boards now using the BoM from your link. 
Last week I got a delivery of more resistors and capacitors as the original list were out of stock on LCSC.
These are what ive ordered:
https://www.lcsc.com/product-detail/_UNI-ROYAL-Uniroyal-Elec-_C57436.html
https://www.lcsc.com/product-detail/_STE-Songtian-Elec-_C366821.html
https://www.lcsc.com/product-detail/_UNI-ROYAL-Uniroyal-Elec-_C131702.html

Now the confusing bit - if I solder in a board with the LLC and the 10k resistor everything works.
When I add the capacitors and resistors nothing.

Ive tried varying the distance from the capacitor to the board, rotating the resistors, changing solder, changing temperature...
Nothing works.

Any ideas?
Hardware test wise they all show intermittent errors yet my soldering is the same on each.

No GAP on capacitors:
![no gap](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/6a5a37af-4d3b-4ab3-ad6d-89ee36a9c592)
![no gap overviewIMG_2363](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/f5da8029-5e86-4a6a-91b7-f87c3ae2382a)
![no gap test](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/5016a621-4390-4b63-ab9b-167822ac6f51)

GAP on capacitors:
![gap 2](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/0ad584f7-fa40-4729-a5ea-2918eaddd65f)
![gap](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/fd00fbbb-7284-43b4-9567-c94be2243b1a)
![IMG_2361](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/620f37c8-78e6-41fd-a1c3-13a2c437d4f3)

**Comment by visualapproach** [comment]
The hw test is testing at a fairly high rate which maybe gets filtered out by the low pass filter. In the function'handlehwtest' (IIRC) try increasing the delaymicroseconds from 10 to 50 on both places.

**Comment by metaljay** [comment]
changed to 50 microseconds as above, still getting the same error, even if I change to 200 microseconds its the same result:

CIO to DSP pin 3 fail!
CIO to DSP pin 4 fail!
CIO to DSP pin 5 fail!

DSP to CIO pin 3 fail!
DSP to CIO pin 4 fail!
DSP to CIO pin 5 fail!

Ive checked my soldering all looks good. Could it be that the resistors are larger than they should be?

![Untitled](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/44ba992d-cecb-4250-bdd2-fe927a607e26)

**Reply by visualapproach** [reply]
The size of the resistors is not important. The resistance is!

**Reply by visualapproach** [reply]
If it passes the test on the A ports you know it's the filters that is the problem. Now, the capacitors would still be able to mess with the results if they are shorted or something.

**Comment by metaljay** [comment]
sorry I meant size (amount) of the resistance not the physical size.
Ive measured with a multimeter and its ≈557ohms across each resistor so they seem fine.
measuring the capacitors seems odd as I get different readings across each capacitor I measure.

Ill solder again on the 'A' ports just to double check I wasnt going mad...

**Comment by metaljay** [comment]
If I make up a board fresh and just install the LLC and the 10k resistor all works fine.
I can’t for the life of me understand what is going wrong with the resistors / capacitors. 

how can I try determine what's failing? will the board still work if I only solder in the capacitors? (and vice versa). Guessing ill need to bridge the gap?

![IMG_2374](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/35b984ed-9abf-41e0-bf66-9850b4ecc8dc)
![IMG_2373](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/a6aa932a-a37d-4b1b-bd14-85faaf4ec9c8)

**Reply by wyliecoyoteuk** [reply]
Were they from China? they may be rejects, or even fakes.
Capacitors shouldn't pass DC only AC, so they should read as high resistance or open circuit on a Multimeter (they should read low and then go to open circuit once they have charged up)

**Reply by metaljay** [reply]
Links above, normal parts from LCSC. Certainly not fakes or rejects

**Comment by visualapproach** [comment]
> how can I try determine what's failing? will the board still work if I only solder in the capacitors? (and vice versa). Guessing ill need to bridge the gap?


The Caps are connected between A-ports and ground. Without resistors the B ports will be disconnected. With bridges (0 Ohm resistors) the A-ports and B-ports will be the same. With real resistors the B port inputs will charge the caps through the resistors. This will take a short while before a logic HIGH is read on the input. Meaning high frequence noise will be filtered out. The R and C values determine at what frequency this cutoff will occur.

**Reply by metaljay** [reply]
Sorry to sound stupid, but I’m not sure how to read this, are you saying I can install the caps then bridge the resistors (with a wire)? 

I’m trying to eliminate one or the other.

**Reply by visualapproach** [reply]
I wasn't telling you what to do, I tried to explain how it works and what effect different actions would have.
To find out what is wrong, I'd start with only esp and LLC on the board. Put cables in all 4 ports.
Connect and Check A ports. If all okay, put one cap somewhere on the cio side and do the test again (A ports)
If ok put a resistor next to the cap. 
Test.
If ok connect the B port where you have your cap+res to the other A port (cio to dsp)
Test. Expect one good wire and the rest to fail.
If ok put cap+res on corresponding place on dsp side. Connect B to B. Test
You should have good result on that wire and fail on the rest since they are not connected with resistors.

Now, if you have already soldered all the boards you could test bypassing the resistors between the ports with wires just to see what happens. It should be the same as using the a ports. No point in making such a device from the beginning since it would be the same as using the a ports without resistors. 
Note that you must not bypass the 10 K res to the LLC.

**Reply by metaljay** [reply]
I’m now at a total loss - I’ve built a board from the beginning and done exactly as you’ve written, tested each time and all was successful. Finally soldered all hardware in and all but one pin tested successful. I’m now convinced there is a fault in the PCB manufacture that’s causing my issues on the other boards. How have I had such a high failure rate? I’ve made 12 from this batch and 8 have failed! (Or something else?)

![IMG_2402](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/e0c6a5f0-46d3-4ed4-a0c0-77d492ccdc3b)

**Comment by visualapproach** [comment]
Try delay(5) instead of delaymicroseconds, to see if that changes anything

**Reply by metaljay** [reply]
no change - still all failing on test :(

**Reply by visualapproach** [reply]
Sorry, Can't say why. But if 4 out of 12 works it must be some component(s) that are faulty. Cables/solder/pcb/llc/caps/resistors...
